src/main/java/com/amazon/opendistroforelasticsearch/sql/esdomain/LocalClusterState.java

```diff
@@ -103,6 +113,20 @@ public void setClusterService(ClusterService clusterService) {
         });
     }
 
+    public void setSqlSettings(SqlSettings sqlSettings) {
+        this.sqlSettings = sqlSettings;
+        for (Setting<?> setting : sqlSettings.getSettings()) {
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(
+                setting,
+                newVal -> {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Setting value of [{}] changed to [{}]", setting.getKey(), newVal);
```

Comment:
`"The value of setting [{}] changed to [{}]"`

Reply:
Done!